internal/lsp/source/hover.go

```diff
@@ -93,6 +100,108 @@ func Hover(ctx context.Context, snapshot Snapshot, fh FileHandle, position proto
 	}, nil
 }
 
+func hoverRune(ctx context.Context, snapshot Snapshot, fh FileHandle, position protocol.Position) (*protocol.Hover, error) {
+	ri, err := findRune(ctx, snapshot, fh, position)
+	if err != nil {
+		return nil, nil
+	}
+	rng, err := ri.Range()
+	if err != nil {
+		return nil, err
+	}
+
+	var desc string
+	runeName := runenames.Name(ri.Rune)
+	if runeName[0] == '<' {
+		// It looks like an HTML tag. Trim the surrounding <> characters.
+		// Due to https://github.com/microsoft/vscode/issues/124042,
+		// even when using plaintext, the runename will be hidden by VSCode if
+		// not trimmed from its surrounding <>.
+		runeName = strings.TrimRight(runeName[1:], ">")
+	}
+	if strconv.IsPrint(ri.Rune) {
+		desc = fmt.Sprintf("'%s', U+%04X, %s", string(ri.Rune), uint32(ri.Rune), runeName)
+	} else {
+		desc = fmt.Sprintf("U+%04X, %s", uint32(ri.Rune), runeName)
+	}
+	return &protocol.Hover{
+		Contents: protocol.MarkupContent{
+			Kind:  snapshot.View().Options().PreferredContentFormat,
+			Value: desc,
+		},
+		Range: rng,
+	}, nil
+}
+
+type runeInfo struct {
+	Rune rune
+	MappedRange
+}
+
+// findRune returns rune information for a position in a file.
+func findRune(ctx context.Context, snapshot Snapshot, fh FileHandle, pos protocol.Position) (*runeInfo, error) {
+	ctx, done := event.Start(ctx, "source.findRune")
```

Comment:
TODO: Ask the maintainers if that should be kept. 

`source.FindIdentifier`  does this, and since we're also traversing the AST, it makes sense to do it as well here.